File: backend/src/scripts/seed.js
import bcrypt from "bcrypt";
import "dotenv/config";
import { poolCentral } from "../config/db.js";

async function seed() {
  const email = "admin@benjeddou-erp.ma";
  const motDePasse = "Test1234!";
  const hash = await bcrypt.hash(motDePasse, 10);

  await poolCentral.query(
    `INSERT INTO super_admin (nom, email, mot_de_passe)
     VALUES ($1, $2, $3)
     ON CONFLICT (email) DO UPDATE SET mot_de_passe = EXCLUDED.mot_de_passe`,
    ["Admin Test", email, hash]
  );

  console.log("Compte de test créé :");
  console.log("  email :", email);
  console.log("  mot de passe :", motDePasse);
  process.exit(0);
}

seed().catch((err) => {
  console.error(err);
  process.exit(1);
});

File: backend/app/scripts/seed.py
import bcrypt
from dotenv import load_dotenv
load_dotenv()

from app.db import pool_central

EMAIL = "admin@benjeddou-erp.ma"
MOT_DE_PASSE = "Test1234!"


def seed():
    hash_ = bcrypt.hashpw(MOT_DE_PASSE.encode("utf-8"), bcrypt.gensalt()).decode("utf-8")

    conn = pool_central.getconn()
    try:
        with conn.cursor() as cur:
            cur.execute(
                """
                INSERT INTO super_admin (nom, email, mot_de_passe)
                VALUES (%s, %s, %s)
                ON CONFLICT (email) DO UPDATE SET mot_de_passe = EXCLUDED.mot_de_passe
                """,
                ("Admin Test", EMAIL, hash_),
            )
        conn.commit()
    finally:
        pool_central.putconn(conn)

    print("Compte de test créé :")
    print("  email :", EMAIL)
    print("  mot de passe :", MOT_DE_PASSE)


if __name__ == "__main__":
    seed()
